### DISCUSSION TITLE
Is there a way to update a fork with changes in the upstream repo?

### DISCUSSION THREAD

**Discussion by catostrophe** [renovatebot/renovate#26696] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.137.3

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

We have a few projects that are organized in the following way:
- an upstream repo with the common code - an app skeleton
- a bunch of forks with customization - actual apps

We can't pack the common code as a lib and use a dependency manager due to multiple reasons.

When we change the common code, we need to update forks from the upstream in dozens of projects.

We'd like to automate this with the Renovate bot. Is there a way to do this?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by viceice** [comment, marked as answer]
no

**Reply by catostrophe** [reply]
Thanks for the reply. Is there at least a way to introduce this functionality in Renovate? Can I submit a feature request?

**Reply by rarkins** [reply]
It sounds like you're after an automatic for fork sync system?

**Reply by catostrophe** [reply]
Yes, we use Renovate for updating java, dotnet, js, gitlab, docker, etc., including git-submodules.

I was thinking about using Renovate for fork syncing also if possible. Otherwise, we'll need another tool.

**Reply by rarkins** [reply]
Renovate is geared towards updating _files_ in the repo, so that's why for example git-submodules makes sense. But I think that force pushing a new SHA to branches to the repo doesn't seem to fit
